code/modules/surgery/bodyparts/dismemberment.dm

```diff
@@ -83,7 +129,29 @@
 	update_limb(1)
 	C.bodyparts -= src
 	if(held_index)
-		C.unEquip(owner.get_item_for_held_index(held_index), 1)
+		var/obj/item/I = owner.get_item_for_held_index(held_index)
+		var/did_special_unequip = FALSE
+		if(istype(I, /obj/item/weapon))
+			var/obj/item/weapon/W = I
+			if(W.attach_on_dismember)
+				attached_weapon = W
+				attached_weapon.forceMove(src)
+
+				// This is kind of horrible and I apologize. 
+				// TODO: allow new_loc on unEquip
```

Comment:
as discussed, *something* should be done about this, but not 100%
Due to issue with items deleting themselves on drop, etc.
you could add a new arg to unEquip, "callDropped = TRUE" and then call unEquip here with that as false, I suppose.